src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,55 +1,94 @@
 import {fetchAllMatchesForAGroup} from "../regExpUtils";
 import builtInPredicates from "./builtInPredicates";
 
-export function removeFileExtension(path: string) {
-	const fetchFileExtension = /(.*)\.[^\.]*/;
-
+/**
+	* @param path - The path to be curated
+	* @return The path without its extension (removing the last dot and everything after)
+	*/
+export function removeFileExtension(path: string): string {
+	const fetchFileExtension = /(.+)\.[^\.]*/;
 	const fetch: string[] | null = fetchFileExtension.exec(path);
 	if (fetch === null) { return ""; }
 	return fetch[1];
 }
 
-export function removeModuleDeclarations(prologProgram: string) {
+/**
+	* Necessary because module declarations break tau-prolog
+	* @param path - The program code to be curated
```

Comment:
*Räusper* if you copy-paste, then do it correctly! :D 
The parameter is called `prologProgram` in this function and the following ones...